docs/AQL/master03-syntax.adoc

```diff
@@ -512,42 +517,169 @@ FROM
+The `AVG` function returns the average value of given _expression_ argument. The syntax is `AVG(expression)`.
 
-If there are no matching rows, `AVG()` returns `NULL`.
+If there are no matching rows, then this function returns `NULL`.
 Input values type should be either Integer of Real, and it will also determine the return type.
 
+=== String functions
+
+AQL is providing single-row functions for examining and manipulating string values.
+
+The table below shows the built-in AQL string functions:
+
+[width="100%",cols="1,5",options="header",]
+|===
+|Function       |Description
+
+|LENGTH()       |Returns the length of a string
+|CONTAINS()     |Validates if a string contains other string
+|POSITION()     |Returns the position of the first occurrence of substring in a string
+|SUBSTRING()    |Extracts the substring as specified
+|CONCAT()       |Returns the string that results from concatenating the arguments
+|CONCAT_WS()    |Returns the string that results from concatenating the arguments with separator
+|===
+
+For functions that operate on string positions, the first position is numbered 1.
+For functions that take length arguments, noninteger arguments are rounded to the nearest integer.
+
+==== LENGTH
+
+The `LENGTH` function returns the number of characters of a given _expression_ argument.
+The syntax is `LENGTH(expression)`, where the argument is of type String. The result type is Integer.
+
+==== CONTAINS
+
+The `CONTAINS` function with syntax is `CONTAINS(string, substring)` returns the `true` if the `string` argument expression contains the `substring`.
+The arguments are of type String, while the result type is Boolean.
+
+==== POSITION
+
+The `POSITION` function with syntax is `POSITION(substring, string)` returns the position (index) of the first occurrence of `substring` in `string` argument expression, or zero if it's not present.
+The arguments are of type String, while the result type is Integer.
+
+==== SUBSTRING
+
+The `SUBSTRING` function with syntax is `SUBSTRING(string, position, length)` extracts the substring of `string`, starting at the `position`-th character, and stopping after `length` characters, if that is specified.
+The first argument is of type String, while `position`, `length` are Integers. The result type is String.
+
+The `length` argument is optional, and when not provided this function will extract the substring from the `position` until the end of the given `string` argument.
+
+==== CONCAT
+
+The `CONCAT` function with syntax is `CONCAT(string1, string2, ...)` returns the string that results from concatenating all the arguments. It may have one or more expression arguments, all of type String.
+
+==== CONCAT_WS
+
+The `CONCAT_WS` function with syntax is `CONCAT_WS(separator, string1, string2, ...)` returns the string that results from concatenating all the arguments with `separator` string, i.e. the separator is added between the strings to be concatenated.
+
+=== Numeric functions
```

Comment:
Similarly, we could add these functions to the relevant numeric types - https://specifications.openehr.org/releases/BASE/latest/foundation_types.html#_primitive_types

Reply:
Same thing as above - I checked them, but I was not sure what else should I consider besides the one already in the specs. Happy to discuss more about this.